src/eventHubMessageExplorer.ts

```diff
@@ -0,0 +1,99 @@
+'use strict';
+import * as vscode from 'vscode';
+import { Utility } from './utility';
+import { clientFromConnectionString } from 'azure-iot-device-http';
+import { Message, Client } from 'azure-iot-device';
+import { AppInsightsClient } from './appInsightsClient';
+import { BaseExplorer } from './baseExplorer'
+import Promise = require('bluebird');
+import { Client as EventHubClient, Sender as EventHubSender, Receiver as EventHubReceiver } from 'azure-event-hubs';
+
+export class EventHubMessageExplorer extends BaseExplorer {
+    private _eventHubClient;
+    private _eventHubConnectionstringKey = "eventHubConnectionString";    
+    private _eventHubConnectionStringTitle = "Event Hub Connection String";
+    private _eventHubPathKey = "eventHubPath";
+    private _eventHubPathTitle = "Event Hub Path";
+    private _monitorLabel = "MonitorEventHub";
+    private _messageLabel = 'MessageToEventHub';
+
+    public sendMessageToEventHub(): void {
+        let eventHubConnectionString = Utility.getConfig(this._eventHubConnectionstringKey, this._eventHubConnectionStringTitle);
+        let eventHubPath = Utility.getConfig(this._eventHubPathKey, this._eventHubPathTitle);
+        if(!eventHubConnectionString || !eventHubPath) {
+            this.outputLine(this._messageLabel, 'Invalid arguments of event hub.');
+        }
+        vscode.window.showInputBox({ prompt: 'Enter message to send to event hub' }).then((message: string) => {
+            if (message !== undefined) {
+                try {
+                    let client = EventHubClient.fromConnectionString(eventHubConnectionString, eventHubPath);
+                    let receiveAfterTime = Date.now() - 5000;
+                    client.open()
+                        .then(client.getPartitionIds.bind(client))
+                        .then(() => client.createSender())
+                        .then(this.sendToEventHub(message, this._messageLabel))
+                        .then(this.sendToEventHubDone(client, this._messageLabel))
+                        .catch(this.printError(this._outputChannel, this._messageLabel, client));
+                }
+                catch (e) {
+                    this.outputLine(this._messageLabel, e);
+                }
+            }
+        });
+    }
+
+    public startMonitorEventHubMessage(): void {

```

Comment:
startMonitorEventHubMessage is similar to startMonitorIoTHubMessage. Could we extract a shared function to BaseExplorer which could be shared in startMonitorEventHubMessage  and startMonitorIoTHubMessage?